client/deployments/client.go

```diff
@@ -367,6 +370,24 @@ func (c *Client) UploadArtifact(
 		return errors.Wrap(err, "Cannot read artifact file stats")
 	}
 
+	tr := tar.NewReader(artifact)
+	versionH, err := tr.Next()
+	if err != nil {
+		return errors.Wrap(err, "Cannot find artifact")
+	} else if versionH.Name != "version" {
+		return errors.New("Invalid artifact format")
+	}
+	v := struct {
+		Format string `json:"format"`
+	}{}
+	err = json.NewDecoder(tr).Decode(&v)
+	if err != nil || v.Format != "mender" {
+		return errors.New("Invalid artifact format")
+	}
+	_, err = artifact.Seek(0, io.SeekStart)
+	if err != nil || v.Format != "mender" {
+		return err
+	}
```

Comment:
Could you also reuse the format checker here?
```suggestion
	if err = checkArtifactFormat(artifact); err != nil {
		return err
	}
```